File: play/fetch.rb
#!/usr/bin/env ruby

require 'date'
require "RRDtool"

name    = "minmax"
rrdname = "randome.rrd"
start   = Time.now.to_i

rrd = RRDtool.new(rrdname)
puts "created new RRD database -> #{rrd.rrdname}"

# ---------------------------------------------
#           Update
# ---------------------------------------------
puts " .. fetching #{rrdname}"
(start, stop, names, data) = rrd.fetch(["AVERAGE"])
puts ">Names .."
names.each do  |name|
    puts " --> #{name}"
end
puts ">Data .."
t = start
data.each do  |val|
    puts " #{DateTime.new(t)} --> #{val}"
    t += 300  # know this from elsewhere
end

File: play/fetch.pl
#! /usr/bin/perl

use lib qw( /usr/local/rrdtool-1.2.10/lib/perl );

use RRDs;
my $start=time;
my $rrd="randome.rrd";
my $name = $0;
$name =~ s/\.pl.*//g;

my ($start,$step,$names,$array) = RRDs::fetch $rrd, "AVERAGE";
$ERROR = RRDs::error;
print "ERROR: $ERROR\n" if $ERROR ;
print "start=$start, step=$step\n";
print "                    ";
map {printf("%12s",$_)} @$names ;
print "\n";
foreach my $line (@$array){
    print "".localtime($start),"   ";
    $start += $step;
    foreach my $val (@$line) {
        printf "%12.1f", $val;
    }
    print "\n";
}

